tests/simplex/mapping_fe_fields_02.cc

```diff
@@ -0,0 +1,123 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Like mapping_fe_fields_01 but for deformed meshes tested for linear and
+// quadratic mapping.
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping_fe.h>
+#include <deal.II/fe/mapping_fe_field.h>
+
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <deal.II/simplex/fe_lib.h>
+#include <deal.II/simplex/grid_generator.h>
+#include <deal.II/simplex/quadrature_lib.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+template <int dim>
+class Solution : public Function<dim>
+{
+public:
+  Solution()
+    : Function<dim>(dim)
+  {}
+
+  double
+  value(const Point<dim> &point, const unsigned int compontent) const
+  {
+    return std::sin(point[compontent] * 0.5 * numbers::PI);
+  }
+};
+
+void
+test(const unsigned int mapping_degree)
+{
+  const int dim = 2;
+
+  Triangulation<dim> tria;
+  GridGenerator::subdivided_hyper_cube_with_simplices(tria, 4);
+
+  Simplex::FE_P<dim> fe(mapping_degree);
+  FESystem<dim>      euler_fe(fe, dim);
+
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+
+  DoFHandler<dim> euler_dof_handler(tria);
+  euler_dof_handler.distribute_dofs(euler_fe);
+
+  Vector<double> euler_vector(euler_dof_handler.n_dofs());
+
+  // TODO: not working (missing mapping)
+  // VectorTools::get_position_vector(euler_dof_handler, euler_vector);
+
+  MappingFE<dim> mapping_interpolation(Simplex::FE_P<dim>(1));
+  VectorTools::interpolate(mapping_interpolation,
+                           euler_dof_handler,
+                           Solution<dim>(),
+                           euler_vector);
+
+  MappingFEField<dim> mapping(euler_dof_handler, euler_vector);
+
+  Simplex::QGauss<dim> quadrature_formula(1);
+
+  FEValues<dim> fe_values(mapping,
+                          fe,
+                          quadrature_formula,
+                          update_values | update_gradients);
+
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      fe_values.reinit(cell);
+    }
```

Comment:
Is this really needed? `DataOut::build_patches` should also call this function (using `update_values` and `update_quadrature_points` as arguments).